### PR TITLE
Sign ras-rm-pr-bot commits with GPG key

### PR DESCRIPTION
**Pull request by pricem14pc**
# What and why?

The ras-rm-pr-bot commits that bump versions are not verified. This means we cannot enforce signed commits on the main branch.

This PR introduces a method of signing the BOT commits.

It uses a GPG signing key which is passphrase protected, stored as Github Actions secrets. It does not expose the key or passphrase in the Github Actions output.

Existing BOT secrets
```
BOT_EMAIL
BOT_TOKEN
BOT_USERNAME
```
New BOT secrets
```
BOT_GPG_KEY
BOT_GPG_KEY_ID
BOT_GPG_KEY_PASSPHRASE
```
Ensure BOT commits are signed and verified, and that the secrets aren't exposed. The build workflow for this PR should see the following successful steps
```
Import BOT GPG key
Prepare gpg CLI signing step
Setup git
```
Before the original `update versions` step successfully bumps the Chart.yaml version and creates a signed commit on this PR branch.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -76,10 +76,28 @@ jobs:
         run: |
           git fetch --tags
           echo "tag=$(git describe --tags --abbrev=0)" >> $GITHUB_ENV
+      - name: Import BOT GPG key
+        run: echo $BOT_GPG_KEY | base64 --decode | gpg --batch --import
```
Comment (pricem14pc) [inline .github/workflows/main.yml:80]: This ensures the key is never persisted outside of the encrypted Github Actions secrets, or the GPG agent.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -76,10 +76,28 @@ jobs:
...
+        env:
+          BOT_GPG_KEY: ${{ secrets.BOT_GPG_KEY }}
+      - name: Prepare gpg CLI signing step
+        run: |
+          rm -rf /tmp/gpg.sh
+          echo '#!/bin/bash' >> /tmp/gpg.sh
+          echo 'gpg --batch --pinentry-mode=loopback --passphrase $BOT_GPG_KEY_PASSPHRASE $@' >> /tmp/gpg.sh
```
Comment (pricem14pc) [inline .github/workflows/main.yml:87]: This allows us to override the default git gpg signing process, and include a passphrase for additional security in this automated flow.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -76,10 +76,28 @@ jobs:
...
+          chmod +x /tmp/gpg.sh
+      - name: Setup git
+        run: |
+          git config commit.gpgsign true
+          git config user.signingkey "${{ secrets.BOT_GPG_KEY_ID }}"
+          git config gpg.program /tmp/gpg.sh
```
Comment (pricem14pc) [inline .github/workflows/main.yml:93]: This is where we override the default signing process to use our own amended CLI options, allowing us to pass in the passphrase.

**Review by anwilkie — APPROVED** [review]

**Review by arroyoAle — APPROVED** [review]
Looks good, pr-bot is verified for this repo and github actions don't expose any secrets in the action logs

**Pull request merged**